### PR TITLE
Update and test the profile screen

### PR DESCRIPTION
**Pull request by nada-ezziti**
Improving the quality of the code and updating the profile screen to match its new design

Putting the text displayed by the help section in the file strings.xml to avoid hardcoded text
Removing the part about the exercises and questions achieved from the profile screen
Changing the colors used to align with the Figma design
Adding the colors to the file colors.xml to avoid hardcoded colors
Adding a file Utils functions that can be reused in multiple screen to avoid duplication
Creation of the function ReusableButtonWithIcon and ReusableTextButton
In the 4th commit :

adding the tests for the Profile Screen
Adding new file for the Screen MyStats
Implementing the navigation to the new screen from the Profile Screen

### CODE REVIEW

**Review by samymessouak — APPROVED** [review]
- Moving hardcoded text and colors to strings.xml and colors.xml aligns with best practices, ensures easier maintenance and localization in the future.
- The update to the ProfileScreen to match the Figma design ensures that the UI remains consistent with the project’s design specifications.
- Adding reusable functions (ReusableButtonWithIcon, ReusableTextButton) and the Utils class to avoid code duplication is a great enhancement.
- The addition of unit tests for the ProfileScreen and the new MyStatsScreen increases test coverage.
- The ProfileScreenTest includes detailed checks for UI elements and interactions, such as buttons being correctly displayed, clickable, and performing actions. These tests seem comprehensive.

Areas to Consider:
_Testing Dialog_:
 -In the test case dialogIsDisplayed_whenHelpBoxIsVisible, it sets the isHelpBoxVisible state outside of the ProfileScreen call, but the test never verifies if the help box is shown without interaction. It might be useful to test the dialog without relying on UI interaction if the state is controlled outside.


Your PR looks solid to me, congrats

File: app/src/main/java/com/github/se/signify/MainActivity.kt [hunk 1]
```diff
@@ -118,11 +118,6 @@ fun SignifyAppPreview() {
             profilePictureUrl = null, // Replace with actual URL or null
```
Comment (ThomasMPont) [inline app/src/main/java/com/github/se/signify/MainActivity.kt:118]: This is way too many arguments for a screen. I think it should be passed as a view model or some class.

**Review by NBCCF025 — APPROVED** [review]
### Pull Request: **Profile Screen Update and Code Quality Improvements**

#### Review Summary:
This pull request brings significant improvements to the Profile Screen UI. The refactor focuses on removing hardcoded values, aligning the design with Figma, and creating reusable components. These are all excellent steps toward cleaner, more maintainable code.

#### Positive Aspects:
1. **Separation of Strings and Colors**:
   - Moving the help section text to `strings.xml` and the colors to `colors.xml` is a best practice that promotes maintainability and makes localization easier. This avoids hardcoded values and enhances the readability of the code.
   
2. **UI Alignment with Design**:
   - Updating the Profile Screen to match the new Figma design ensures consistency between design and implementation. It's good to see that the color palette was updated in the app to reflect the design accurately.

3. **Reusable Components**:
   - The creation of `ReusableButtonWithIcon` and `ReusableTextButton` is a great step toward reducing duplication. By introducing reusable UI components, future changes can be made in a single place, improving flexibility and maintainability.

4. **Utils File**:
   - Adding a `Utils` file for common functions is a solid approach for promoting code reuse across multiple screens.

#### Suggested Improvements:

1. **Documentation for Reusable Components**:
   - While the new reusable components (`ReusableButtonWithIcon`, `ReusableTextButton`) are a great addition, it would be helpful to add comments or documentation around their intended use. This ensures that other team member can easily understand how and when to use them.

2. **Tests for Utils Functions**:
   - Since you’ve added a `Utils` file, consider adding unit tests for these functions to ensure that the reusable logic works correctly across different use cases. This will help catch any edge cases early.

3. **Code Formatting**:
   - Ensure consistent formatting across all files. For instance, double-check spacing and indentation, especially around the newly added functions. Following the project’s existing style guide will maintain a uniform look throughout the codebase. You can use "CTRL + ALT + L".


4. **Testing Profile Screen**:
   - As mentioned in a comment on this specific file, you can make a function to set default value to the profilescreen for the setContent in order to call it whitout repeating each time the complete call to profilescreen.

---

### Summary
All changements contribute positively to the project’s. I approve this pull request.

File: app/src/main/java/com/github/se/signify/Utils.kt [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+              width = 2.dp, brush = SolidColor(colorResource(R.color.dark_gray))),
+      colors = ButtonDefaults.buttonColors(colorResource(R.color.blue)),
+      modifier = Modifier.fillMaxWidth().height(40.dp).testTag(text + "Button")) {
+        Text(
+            text,
+            fontWeight = FontWeight.Bold,
+            color = colorResource(R.color.dark_gray),
+            fontSize = 16.sp)
+      }
+}
```
Comment (samymessouak) [inline app/src/main/java/com/github/se/signify/Utils.kt:54]: It seems better to pass all design caracteristics of the button as argument to have a more flexible function.
As i worked with your class i've done it.

**Review by ThomasMPont — COMMENTED** [review]
Good job with the updated profile screen.
I see this branch has just been merged, but I still wanted to add a few comments.

## Tests
Since SonarCloud test reports aren't working yet, I ran a test report locally.
I can confirm the new screen reaches over 80% instruction coverage, congrats!

## Figma 
Parts of this screen doesn't match its design on Figma. Since the branch has been merged already, and its reviewers let it slide. I suggest we try to fix the look of the screen at a later date. Ideally, elements repeated between screens, like the help button, will be a common composable shared between screens in the future.

## Commits
Your commits could most probably be split into several smaller commits. Try to stick to one commit per file at the very least, unless changes depend on each other.
Remember to follow universal commit message guidelines, such as using the imperative form and capitalizing each commit.
